File: abc364a/abc364a.ml
open Printf
open Scanf

let solve n ls =
  let f (x, y) s =
    if y = 2 then (x, y)
    else if s = "sweet" then (x + 1, y + 1)
    else (x + 1, 0) in
  let (ct , _) = List.fold_left f (0, 0) ls in
  if ct = n then "Yes" else "No"

let () =
  let id x = x in
  scanf "%d " (fun n ->
    List.init n (fun _ -> scanf "%s " id) |> solve n) |> printf "%s\n"

File: abc364a/abc364a.hs
{-# LANGUAGE OverloadedStrings #-}
import Control.Monad (replicateM)
import Data.Bool (bool)

main :: IO ()
main = do
  n <- readLn
  solve n <$> replicateM n getLine >>= putStrLn

solve :: Int -> [String] -> String
solve n = bool "No" "Yes" . (== n) . fst . foldl f (0, 0)
  where
    f (ct, sw) s
      | sw == 2 = (ct, sw)
      | s == "sweet" = (ct + 1, sw + 1)
      | otherwise = (ct + 1, 0)
